### PR TITLE
docs: update note on annotations

### PR DESCRIPTION
**Pull request by wyardley**
#### What this PR does / why we need it:
Use less gendered language, and slightly rework wording, in some notes
on pod / service annotations

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated

### CODE REVIEW

File: charts/datadog/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: datadog
-version: 2.35.4
+version: 2.35.5
```
Comment (wyardley) [inline charts/datadog/Chart.yaml:3]: don't think this is really needed, but the CI checks wanted a version bump. Happy to take it back out if not necessary.

File: charts/datadog/templates/agent-cilium-network-policy.yaml [hunk 2]
```diff
@@ -152,11 +152,11 @@ specs:
               - port: "{{ $.Values.datadog.apm.port }}"
                 protocol: TCP
 {{- end }}
-# The agents are susceptible to connect to any pod
-# that would be annotated with auto-discovery annotations.
+# The agents are susceptible to an issue connecting to any pod that
+# is annotated with auto-discovery annotations.
```
Comment (wyardley) [inline charts/datadog/templates/agent-cilium-network-policy.yaml:156]: adjusted the wording on these 4 again; let me know if this doesn't accurately describe the issue and / or needs additional improvements.

**Review by jtappa — APPROVED** [review]
❤️ thank you from docs team!

**Comment by wyardley** [comment]
@clamoriniere do you want this one with, or without changelog update / version bump. I think the lint errors will fail without it

**Comment by jtappa** [comment]
Hi @wyardley, someone from the containers team has been pinged, hopefully they review shortly! Thanks for your contribution!

**Review by L3n41c — APPROVED** [review]

**Pull request merged**